go/vt/vtgate/vtgateconn/vtgateconn.go

```diff
@@ -171,6 +171,11 @@ func (sn *VTGateSession) Prepare(ctx context.Context, query string) ([]*querypb.
 	return fields, paramsCount, err
 }
 
+// CloseSession closes the session provided by rolling back any active transaction.
+func (sn *VTGateSession) CloseSession(ctx context.Context) error {
+	return sn.impl.CloseSession(ctx, sn.session)
+}
+
```

Comment:
We should have tests for this to describe why this is needed. We do run stuff like https://go.dev/blog/deadcode on the codebase to clean up unused parts and it would identify this and we're remove it again. 

Making sure it's at least tested and described in the tests why it's needed would be required to at least try to avoid it being removed again in the future.

Reply:
OK. I added a bunch of tests for this to the framework.

Took a bit of wrangling for the existing tests and the fake server to track "active txns" to make this work, but it is now tested.